plugins/workflow/plugin.go

```diff
@@ -40,38 +43,27 @@ func (svc *Plugin) Init(temporal temporal.Temporal, server server.Server, log lo
 
 // Serve starts workflow service.
 func (svc *Plugin) Serve() chan error {
-	errCh := make(chan error, 1)
-
-	pool, err := NewWorkflowPool(context.Background(), svc.server)
-	if err != nil {
-		errCh <- errors.E(errors.Op("newWorkflowPool"), err)
-		return errCh
-	}
+	svc.mu.Lock()
+	defer svc.mu.Unlock()
```

Comment:
We don't need mutex here. Serve function maintained by Endure which has the protection of this func.